diff --git a/node_modules/@wuchale/svelte/dist/transformer.js b/node_modules/@wuchale/svelte/dist/transformer.js
index b377e3d..843f11e 100644
--- a/node_modules/@wuchale/svelte/dist/transformer.js
+++ b/node_modules/@wuchale/svelte/dist/transformer.js
@@ -220,7 +220,7 @@ export class SvelteTransformer extends Transformer {
         return txts;
     }
     visitIfBlock(node) {
-        const txts = this.visit(node.test);
+        const txts = this.inScopeVisit({ type: 'expression' }, node.test);
         txts.push(...this.visitFragment(node.consequent, false));
         if (node.alternate) {
             txts.push(...this.visitFragment(node.alternate, false));
@@ -228,7 +228,7 @@ export class SvelteTransformer extends Transformer {
         return txts;
     }
     visitEachBlock(node) {
-        const txts = [...this.visit(node.expression), ...this.visitFragment(node.body, false)];
+        const txts = [...this.inScopeVisit({ type: 'expression' }, node.expression), ...this.visitFragment(node.body, false)];
         if (node.key) {
             txts.push(...this.visit(node.key));
         }
@@ -238,10 +238,10 @@ export class SvelteTransformer extends Transformer {
         return txts;
     }
     visitKeyBlock(node) {
-        return [...this.visit(node.expression), ...this.visitFragment(node.fragment, false)];
+        return [...this.inScopeVisit({ type: 'expression' }, node.expression), ...this.visitFragment(node.fragment, false)];
     }
     visitAwaitBlock(node) {
-        const txts = this.visit(node.expression);
+        const txts = this.inScopeVisit({ type: 'expression' }, node.expression);
         if (node.then) {
             txts.push(...this.visitFragment(node.then, false));
         }
