diff --git a/node_modules/wuchale/dist/adapter-utils/mixed-visitor.js b/node_modules/wuchale/dist/adapter-utils/mixed-visitor.js
index e212611..e143bab 100644
--- a/node_modules/wuchale/dist/adapter-utils/mixed-visitor.js
+++ b/node_modules/wuchale/dist/adapter-utils/mixed-visitor.js
@@ -266,7 +266,10 @@ export class MixedVisitor {
             }
             else if (props.commentDirectives.forceType !== false) {
                 if (this.#props.leaveInPlace(child)) {
+                    this.#mod.set(scope.type, newMod());
                     txts.push(...this.#props.visitFunc(child));
+                    txts.push(...this.#applyModClear(scope.type));
+                    this.#mod.set(scope.type, mod);
                 }
                 else if (this.#props.isExpression(child)) {
                     txts.push(...this.#props.visitFunc(child));
